controller/node_controller.go

```diff
@@ -995,6 +1001,169 @@ func (nc *NodeController) syncNodeStatus(pod *corev1.Pod, node *longhorn.Node) e
+			fmt.Sprintf("Get namespace executor failed: %v", err))
+		return
+	}
+	args := []string{"show", "status"}
+	if result, _ := nsexec.Execute(nil, "multipathd", args, lhtypes.ExecuteDefaultTimeout); result != "" {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonMultipathdIsRunning),
+			fmt.Sprintf("Multipathd is running on %v known to have a breakage that affects Longhorn. See description and solution at https://longhorn.io/kb/troubleshooting-volume-with-multipath", node.Name))
+		return
+	}
+
+	node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeMultipathd, longhorn.ConditionStatusTrue, "", "")
+}
+
+func (nc *NodeController) syncNFSClientVersion(kubeNode *corev1.Node, node *longhorn.Node, namespaces []lhtypes.Namespace) {
+	kernelVersion := kubeNode.Status.NodeInfo.KernelVersion
+	nfsClientVersions := []string{"CONFIG_NFS_V4_2", "CONFIG_NFS_V4_1", "CONFIG_NFS_V4"}
+
+	nsexec, err := lhns.NewNamespaceExecutor(lhtypes.ProcessNone, lhtypes.HostProcDirectory, namespaces)
+	if err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonNamespaceExecutorErr),
+			fmt.Sprintf("Get namespace executor failed: %v", err))
+		return
+	}
+
+	kernelConfigPath := "/boot/config-" + kernelVersion
+	args := []string{kernelConfigPath}
+	if _, err := nsexec.Execute(nil, "ls", args, lhtypes.ExecuteDefaultTimeout); err != nil {
+		node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusFalse,
+			string(longhorn.NodeConditionReasonKernelConfigIsNotFound),
+			fmt.Sprintf("Failed to check %v, because %v does not exist on node %v: %v", nfsClientVersions, kernelConfigPath, node.Name, err))
+		return
+	}
+
+	for _, ver := range nfsClientVersions {
+		args := []string{ver + "=", kernelConfigPath}
+		result, err := nsexec.Execute(nil, "grep", args, lhtypes.ExecuteDefaultTimeout)
+		if err != nil {
+			logrus.WithError(err).Warnf("Failed to find kernel config %v on node %v", ver, node.Name)
+		}
+		enabled := strings.TrimSpace(strings.Split(result, "=")[1])
+		switch enabled {
+		case "y":
+			node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusTrue, "", "")
+			return
+		case "m":
+			opts := []string{"|grep", "nfs"}
+			if _, err := nsexec.Execute(nil, "lsmod", opts, lhtypes.ExecuteDefaultTimeout); err == nil {
+				node.Status.Conditions = types.SetCondition(node.Status.Conditions, longhorn.NodeConditionTypeNFSClientInstalled, longhorn.ConditionStatusTrue, "", "")
+				return
+			}
+		default:
+			logrus.Warnf("Unknown kernel config value for %v: %v", ver, enabled)
```

Comment:
Change Warnf to Trancef or Debugf, or it will lead to flooding warning messages

Reply:
Modified to `Debugf`. Thanks.